source/hp/dof_handler.cc

```diff
@@ -1586,6 +1589,10 @@ namespace hp
     dealii::internal::hp::DoFHandlerImplementation::Implementation::
       reserve_space(*this);
 
+    // initialize all p-refinement and p-coarsening flags.
+    // important: do this after flushing levels in reserve_space().
+    create_adaptive_flags();
```

Comment:
We initially reserved space for the p-refinement flags in the `set_fe()` call above.

All `DoFLevels` will be flushed during the `reserve_space()` call, so we somehow need to forward these flags. We could either copy the flags similarly to the `active_fe_indices` in the `reserve_space_release_space` function, or simply ignore all previous flags and re-initialize all flag-related containers (which is the case now).

Either way you decide, I'll have to update the documentation accordingly.

Reply:
Let's deal with it the same way as the active fe flags.